components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/common/SynapsePropertiesHandler.java

```diff
@@ -46,6 +55,13 @@ public boolean handleRequest(MessageContext messageContext) {
         boolean isContentTypeNotSet = false;
         if (headers != null) {
             isContentTypeNotSet = headers.get("Content-Type") == null || headers.get("Content-Type").equals("");
+            if (headers.get(APIMgtGatewayConstants.HOST) != null || !("")
```

Comment:
Shall we use ```StringUtils.isNotEmpty()``` here, instead of checking null and empty separately?

Reply:
Fixed by https://github.com/wso2/carbon-apimgt/pull/9734/commits/65d29928aca59d0422a578d518c60924ed788958